src/tool_writeout_json.c

```diff
@@ -70,7 +70,7 @@ int jsonquoted(const char *in, size_t len,
       break;
     default:
       if(*i < 32)
-        result = curlx_dyn_addf(out, "\\u%04x", *i);
+        result = curlx_dyn_addf(out, "\\u%04x", (unsigned)*i);
```

Comment:
the cast is unnecessary now. however if `*i` was type char (assume `char` is signed) then you'd have to cast to `unsigned char` not `unsigned`
~~~
char x = 'é'; // decimal: -23, hex: e9
printf("%x", (unsigned)x); // ffffffe9
printf("%x", (unsigned char)x); // e9
~~~

Reply:
@jay It is technically necessary because `sizeof (usnigned char) != sizeof (unsigned)` and `curlx_dyn_addf` takes arguments with a variadic list `...`.

Reply:
According to <https://cppreference.com/> casting to `int`/`unsigned` is unnecessary for types smaller than `int` because they get converted automatically.
https://en.cppreference.com/w/cpp/language/variadic_arguments

But I don't think that is specified in `C89`, I can't find it.

Reply:
it's called default argument promotions. for arguments not prototyped by the function there's integer promotions  and float becomes double. note old c standard c89 refers to integer promotions as "integral promotions".

[3.3.2.2 Function calls](https://port70.net/~nsz/c/c89/c89-draft.html#3.3.2.2)

> _If the expression that denotes the called function has a type that does not include a prototype, the integral promotions are performed on each argument and arguments that have type float are promoted to double. These are called the default argument promotions._

[3.2.1.1 Characters and integers](https://port70.net/~nsz/c/c89/c89-draft.html#3.2.1.1)

> _A char, a short int, or an int bit-field, or their signed or unsigned varieties, or an object that has enumeration type, may be used in an expression wherever an int or unsigned int may be used. If an int can represent all values of the original type, the value is converted to an int; otherwise it is converted to an unsigned int. These are called the integral promotions._